homework-assignments-1/task-2.ts

```diff
@@ -0,0 +1,14 @@
+class User {
+    private username: string;
+    private password: string;
+  
+    constructor(username: string, password: string) {
+      this.username = username;
+      this.password = password;
+    }
+  
+    // Public method to validate login, abstracts password checking logic
+    validateLogin(inputPassword: string): boolean {
```

Comment:
Your public function is exposing implementation of checking password. it must be handled internally. 
` public validateLogin(inputPassword: string): boolean {
    // Abstracts the logic of password comparison
    return this.checkPassword(inputPassword);
  }

  // Private method for password comparison
  private checkPassword(inputPassword: string): boolean {
    // Compare the input password with the actual password
    return inputPassword === this.password;
  }`